controllers/chat/ChatController.js

```diff
@@ -462,23 +467,61 @@ module.exports = {
     };
 
     try {
-      if (!members.includes(req.userId)) members.push(req.userId);
       await client.connectUser({id: req.userId}, req.token);
+      let channel = null;
+      // check if member are both anonymous
+      if (!members.includes(req.userId)) members.push(req.userId);
+      let is_anon_to_anon = await is_all_anon_user(members);
+      if (is_anon_to_anon) {
+        // check if channel already exist
+        let new_channel = await handle_anon_to_anon_channel_owner(req.userId, members[0], context);
+        await handle_anon_to_anon_channel_member(new_channel, members[0], oldChannelId, context);
+        channel = client.channel('messaging', new_channel.channel_id, {
+          members
+        });
+      } else {
+        channel = client.channel('messaging', {
+          members
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for creating a chat channel has been updated to handle anonymous-to-anonymous chats. If all members are anonymous, it creates an anonymous-to-anonymous chat channel; otherwise, it creates a regular chat channel.
```diff
-      await client.connectUser({id: req.userId}, req.token);
- 
-      const channel = client.channel('messaging', {
-        members
-      });
+      await client.connectUser({id: req.userId}, req.token);
+      let channel = null;
+      // check if member are both anonymous
+      if (!members.includes(req.userId)) members.push(req.userId);
+      let is_anon_to_anon = await is_all_anon_user(members);
+      if (is_anon_to_anon) {
+        // check if channel already exist
+        let new_channel = await handle_anon_to_anon_channel_owner(req.userId, members[0], context);
+        await handle_anon_to_anon_channel_member(new_channel, members[0], oldChannelId, context);
+        channel = client.channel('messaging', new_channel.channel_id, {
+          members
+        });
+      } else {
+        channel = client.channel('messaging', {
+          members
+        });
+      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->